src/vs/server/node/server.main.ts

```diff
@@ -1011,15 +1011,29 @@ export async function main(options: IServerOptions): Promise<void> {
 		let port = 3000;
 		if (parsedArgs.port) {
 			port = Number(parsedArgs.port);
+			if (isNaN(port)) {
+				logService.error('Port must be a number.');
+				process.exit(1);
+			} else if (!(port >= 0 && port < 65536)) {
+				logService.error('Port out of range.');
+				process.exit(1);
+			}
```

Comment:
I don't think this validation is needed as this is already done by node 
```
./resources/server/web.sh --port 999999
[main 2021-10-02T04:43:58.008Z] rejected promise not handled within 1 second: RangeError [ERR_SOCKET_BAD_PORT]: options.port should be >= 0 and < 65536. Received 999999.
[main 2021-10-02T04:43:58.010Z] stack trace: RangeError [ERR_SOCKET_BAD_PORT]: options.port should be >= 0 and < 65536. Received 699999.
    at validatePort (internal/validators.js:217:11)
    at Server.listen (net.js:1446:5)
```

Reply:
Fair enough, I thought this was better looking than this validation, but I agree it probably makes sense to not add this.